src/data/init-abilities.ts

```diff
@@ -210,12 +211,7 @@ export function initAbilities() {
       PostAttackApplyBattlerTagAbAttr,
       false,
       (_user, target, move) =>
```

Comment:
```suggestion
      (_user, target, move) => {
        const targetStatus = target.getStatusEffect();
        const statusCheck = targetStatus ? ![StatusEffect.FREEZE, StatusEffect.SLEEP].includes(targetStatus) : true;
        if (
          !move.hasAttr(FlinchAttr)
          && !target.turnData.acted
          && move.category !== MoveCategory.STATUS
          && statusCheck
        ) {
          return 10;
        }
        return 0;
      }
```
This way you can keep the same behavior as before even with `Pokemon.status` being `protected`.

Reply:
Ah, the issue is that flinch should still affect the Pokemon even if they're asleep/frozen which is why I deleted the condition. It's an issue for a separate PR.

Reply:
I thought that's what the check was there for?

Reply:
No, the check will prevent Pokemon that are sleeping or frozen from getting flinched. They should still be flinched